### ISSUE TITLE
2.1: Edit and display of SOUR:DATA is imcomplete

### ISSUE THREAD

**Issue by ric2016** [fisharebest/webtrees#4239]
1. When trying to add Data to a source, the edit dialog is empty. add-fact probably requires special code to handle adding new EVEN, AGNC, and NOTE subtags. ([cf the definiton here](https://github.com/fisharebest/webtrees/blob/640069c383a0e7f9043a188577ed5750402c5c76/app/Factories/ElementFactory.php#L693)).

The edit dialog for existing SOUR:DATA also lacks options to add additional EVEN etc.

2. Existing SOUR:DATA is not displayed when viewing a source.

Will SOUR:DATA:* be fully supported in webtrees core?

**Comment by fisharebest** [comment]
The `SOUR:DATA:XXXX` tags are hidden by default., as they are rarely used.

You can enable them globally in: Control panel -> GEDCOM tags.
You can enable them on an edit page using the "Show all GEDCOM tags" option.

Perhaps it would be better if the config option would hide `SOUR:DATA` - not `SOUR:DATA:XXX`

**Comment by ric2016** [comment]
> Perhaps it would be better if the config option would hide SOUR:DATA - not SOUR:DATA:XXX

Yes - It doesn't make sense to have SOUR:DATA without the subtags.
Note: While I can un-hide other tags in the config, unhiding SOUR:DATA:XXX doesn't seem to work: After saving, the checkbox is back.

> Existing SOUR:DATA is not displayed when viewing a source.

Shouldn't this work in any case though? It doesn't.

(Also, why are some tags grouped in the config? CHR has nothing to do with CHRA, why is it not possible to enable them independently? I would also suggest to split up FCOM and CONF, and probably others as well)

**Comment by fisharebest** [comment]
> Existing SOUR:DATA is not displayed when viewing a source.

The fix for #4242 also fixes this issue.

**Comment by ric2016** [comment]
Thanks, looks good now. That resolves all issues, except for

> Perhaps it would be better if the config option would hide SOUR:DATA - not SOUR:DATA:XXX

which isn't that important.
